### DISCUSSION TITLE
dist_train.sh failed in mmdetection 3.x

### DISCUSSION THREAD

**Discussion by vansin** [open-mmlab/mmdetection#8985] [Q&A]
```shell
./tools/dist_train.sh configs/yolact/yolact_r50_8xb8-55e_coco.py 8
/opt/conda/envs/zzs_openmmlab3/lib/python3.8/site-packages/torch/distributed/launch.py:178: FutureWarning: The module torch.distributed.launch is deprecated
and will be removed in future. Use torchrun.
Note that --use_env is set by default in torchrun.
If your script expects `--local_rank` argument to be set, please
change it to read from `os.environ['LOCAL_RANK']` instead. See 
https://pytorch.org/docs/stable/distributed.html#launch-utility for 
further instructions

  warnings.warn(
WARNING:torch.distributed.run:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************
Traceback (most recent call last):
  File "./tools/train.py", line 12, in <module>
    from mmdet.utils import register_all_modules
  File "/data/private/zzs/mmdetection/mmdet/__init__.py", line 10, in <module>
    mmcv_version = digit_version(mmcv.__version__)
AttributeError: module 'mmcv' has no attribute '__version__'
Traceback (most recent call last):
```

**Comment by zhangzscn** [comment]
Package            Version   Editable project location
------------------ --------- -----------------------------
addict             2.4.0
brotlipy           0.7.0
certifi            2022.6.15
cffi               1.15.1
charset-normalizer 2.1.0
click              8.1.3
colorama           0.4.5
coloredlogs        15.0.1
commonmark         0.9.1
contourpy          1.0.5
cryptography       37.0.2
cycler             0.11.0
flatbuffers        2.0
fonttools          4.37.4
humanfriendly      10.0
idna               3.3
importlib-metadata 5.0.0
kiwisolver         1.4.4
Markdown           3.4.1
matplotlib         3.6.0
mkl-fft            1.3.1
mkl-random         1.2.2
mkl-service        2.4.0
mmcv               2.0.0rc1
mmdet              3.0.0rc1  /data/private/zzs/mmdetection
mmengine           0.1.0
model-index        0.1.11
mpmath             1.2.1
numpy              1.23.1
onnxruntime        1.12.1
opencv-python      4.6.0.66
openmim            0.3.2
ordered-set        4.1.0
packaging          21.3
pandas             1.5.0
Pillow             9.2.0
pip                22.2.2
pycocotools        2.0.5
pycparser          2.21
Pygments           2.13.0
pyOpenSSL          22.0.0
pyparsing          3.0.9
PySocks            1.7.1
python-dateutil    2.8.2
pytz               2022.4
PyYAML             6.0
requests           2.28.1
rich               12.6.0
setuptools         65.4.1
six                1.16.0
sympy              1.10.1
tabulate           0.9.0
termcolor          2.0.1
terminaltables     3.1.10
torch              1.12.1
torchaudio         0.12.1
torchvision        0.13.1
typing_extensions  4.3.0
urllib3            1.26.11
wheel              0.37.1
yapf               0.32.0
zipp               3.8.1

**Comment by zhangzscn** [comment]
conda create --name openmmlab python=3.8 -y
conda activate openmmlab
conda install pytorch torchvision torchaudio cudatoolkit=11.6 -c pytorch -c conda-forge
pip install -U openmim
mim install mmengine
mim install "mmcv>=2.0.0rc1"
git clone https://github.com/open-mmlab/mmdetection.git -b dev-3.x
cd mmdetection
pip install -v -e .

**Comment by vansin** [comment]
please try to 
```python
from mmcv import __version__
print(__version__)
```

**Reply by zhangzscn** [reply]
Traceback (most recent call last):
  File "./test.py", line 1, in <module>
    from mmcv import __version__
ImportError: cannot import name '__version__' from 'mmcv' (unknown location)

**Reply by vansin** [reply]
单卡能训练起来吗？
Can single GPU be trained up?

```shell
python tools/train.py xxxx
```

**Reply by zhangzscn** [reply]
python ./tools/train.py configs/yolact/yolact_r50_1xb8-55e_coco.py 
Traceback (most recent call last):
  File "./tools/train.py", line 12, in <module>
    from mmdet.utils import register_all_modules
  File "/data/private/zzs/mmdetection/mmdet/__init__.py", line 10, in <module>
    mmcv_version = digit_version(mmcv.__version__)
AttributeError: module 'mmcv' has no attribute '__version__'

**Reply by zhangzscn** [reply]
不能，出现上面的错误

**Reply by vansin** [reply]
应该是 mmcv 没有装好~

**Reply by zhangzscn** [reply]
这样安装的
pip install -U openmim
mim install mmengine
mim install "mmcv>=2.0.0rc1"

**Reply by vansin** [reply]
please show your `nvidia-smi` and `nvcc -V`

**Reply by zhangzscn** [reply]
Mon Oct 10 17:12:11 2022       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 515.48.07    Driver Version: 515.48.07    CUDA Version: 11.7     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA A40          On   | 00000000:34:00.0 Off |                    0 |
|  0%   54C    P0   136W / 300W |  12503MiB / 46068MiB |     57%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  NVIDIA A40          On   | 00000000:35:00.0 Off |                    0 |
|  0%   58C    P0   156W / 300W |  12531MiB / 46068MiB |     79%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   2  NVIDIA A40          On   | 00000000:36:00.0 Off |                    0 |
|  0%   57C    P0   168W / 300W |  12529MiB / 46068MiB |     61%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   3  NVIDIA A40          On   | 00000000:37:00.0 Off |                    0 |
|  0%   57C    P0   162W / 300W |  12507MiB / 46068MiB |     48%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   4  NVIDIA A40          On   | 00000000:9B:00.0 Off |                    0 |
|  0%   57C    P0   169W / 300W |  12503MiB / 46068MiB |     49%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   5  NVIDIA A40          On   | 00000000:9C:00.0 Off |                    0 |
|  0%   57C    P0   169W / 300W |  12531MiB / 46068MiB |     70%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   6  NVIDIA A40          On   | 00000000:9D:00.0 Off |                    0 |
|  0%   57C    P0   164W / 300W |  12527MiB / 46068MiB |     66%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   7  NVIDIA A40          On   | 00000000:9E:00.0 Off |                    0 |
|  0%   55C    P0   178W / 300W |  12503MiB / 46068MiB |     47%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
+-----------------------------------------------------------------------------+

**Reply by zhangzscn** [reply]
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Thu_Jan_28_19:32:09_PST_2021
Cuda compilation tools, release 11.2, V11.2.142
Build cuda_11.2.r11.2/compiler.29558016_0

**Reply by vansin** [reply]
please try

```shell
conda create -n open-mmlab python=3.8 -y
conda activate open-mmlab
conda install pytorch torchvision -c pytorch
# conda install pytorch torchvision cpuonly -c pytorch
pip install -U openmim
mim install mmengine
mim install "mmcv>=2.0.0rc1"
git clone https://github.com/open-mmlab/mmdetection.git -b dev-3.x
cd mmdetection
pip install -v -e .
```
